torch/csrc/distributed/autograd/context/context.cpp

```diff
@@ -74,7 +74,12 @@ void DistAutogradContext::accumulateGrad(
     it->value().add_(grad);
   } else {
     // First grad for this variable.
-    accumulatedGrads_.insert(variable, grad);
+    if (grad.is_sparse()) {
+      accumulatedGrads_.insert(variable, grad.clone());
```

Comment:
nit: Why does the clone below has a memory format but not this one?

Reply:
This pretty much follows what we do for AccumulateGrad: https://github.com/pytorch/pytorch/blob/6e38a709397ba2299e99011618f78dd5ae4c048a/torch/csrc/autograd/functions/accumulate_grad.cpp#L57. We use contiguous below since the grads are not sparse, but here we don't specify any memory format which defaults to preserving the memory format of the original tensor.

Reply:
But I think AccumulateGrad does that mostly for historical reasons but it is fine to just follow the same behavior.